plugins/modules/server_role.ps1

```diff
@@ -0,0 +1,129 @@
+        $existingServerRole = $server | Get-DbaServerRole -ServerRole $roleName
+    }
+    catch {
+        $module.FailJson("Error checking server role status.", $_.Exception.Message)
+    }
+
+    if ($state -eq "absent") {
+        if ($null -ne $existingServerRole) {
+            try {
+                $removeServerRoleSplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    ServerRole = $roleName
+                    WhatIf = $checkMode
+                    EnableException = $true
+                    Confirm = $false
+                }
+                $output = Remove-DbaServerRole @removeServerRoleSplat
+                if ($output.Status -eq "Success") {
+                    $module.Result.changed = $true
+                }
+                elseif ($output.Status -ne "Success") {
+                    $module.FailJson("Server role [$roleName] was not removed. " + $output.Status)
+                }
+            }
+            catch {
+                $module.FailJson("Deleting server role [$roleName] failed.", $_)
+            }
+        }
+
+        $module.ExitJson()
+    }
+    elseif ($state -eq "present") {
+        # Create server role
+        if ($null -eq $existingServerRole) {
+            try {
+                $newServerRoleSplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    ServerRole = $roleName
+                    WhatIf = $checkMode
+                    EnableException = $true
+                    Confirm = $false
+                }
+
+                if ($null -ne $ownerName) {
+                    $newServerRoleSplat.Owner = $ownerName
+                }
+                $output = New-DbaServerRole @newServerRoleSplat
+                $module.Result.changed = $true
+            }
+            catch {
+                $module.FailJson("Creating server role [$roleName] failed.", $_)
+            }
+        }
+        elseif ($existingServerRole.Owner -ne $ownerName) {
+            $existingServerRole.Owner = $ownerName
+            $existingServerRole.Alter()
+            $output = $existingServerRole
+            $module.Result.changed = $true
+        }
+
+        if ($null -ne $loginName) {
+            try {
+                $invokeQuerySplat = @{
+                    SqlInstance = $sqlInstance
+                    SqlCredential = $sqlCredential
+                    Query = "GRANT ALTER ON LOGIN::[$loginName] TO [$roleName];"
+                    MessagesToOutput = $true
+                    EnableException = $true
+                }
+                $invokeQueryOutput = Invoke-DbaQuery @invokeQuerySplat
+                $module.Result.changed = $true
+            }
+            catch {
+                $module.FailJson("Granting ALTER on role [$roleName] to login [$loginName] failed.", $_)
+            }
+        }
+    }
```

Comment:
@milanjelicic should we remove this part from this module? It can be implemented as another step in the playbook using the generic nonquery module.